app/src/main/java/org/mozilla/fenix/widget/SearchWidgetProvider.kt

```diff
@@ -151,6 +157,22 @@ class SearchWidgetProvider : AppWidgetProvider() {
         }
     }
 
+    private fun RemoteViews.setIcon(context: Context) {
```

Comment:
Consider: 

1. put this in a non-private extension method, in a separate file.
2. promote this to android-components.

Reply:
I have already considered that, but IMO the use case here is very specific ( specific version check, specific vector with problems, specific view) that is why I considered not promoting it to ac or creating an extension function.